ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -697,42 +696,6 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		}
 	}
 
-	if ForcedClusterUpgrade {
-		if len(headPods.Items) == 1 {
-			// head node amount is exactly 1, but we need to check if it has been changed
-			res := utils.PodNotMatchingTemplate(headPods.Items[0], instance.Spec.HeadGroupSpec.Template)
```

Comment:
You can check whether there is any other place referring to this function. If not, we can remove it.

Reply:
`utils.PodNotMatchingTemplate` is used in https://github.com/ray-project/kuberay/blob/a91fc5a217b092b61126945745db777466186029/ray-operator/controllers/ray/utils/util_test.go#L301
so I do not remove it.

Reply:
The test seems to be entirely for `PodNotMatchingTemplate` (https://github.com/ray-project/kuberay/pull/231). Maybe we could also remove it?

Reply:
Done